statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -3229,7 +3229,7 @@ def test_serial_correlation(self, method, lags=None):
                 results = acorr_ljungbox(
                     self.filter_results.standardized_forecasts_error[i][d:],
                     lags=lags, boxpierce=(method == 'boxpierce'),
-                    model_df=self.df_model-1, return_df=False)
+                    model_df=max(self.df_model-1, 0), return_df=False)
```

Comment:
This shouldn't be hardcoded.  It would need to be added as an option for test_serial correlation. Also, isn't df_model always at least 1?  I don't exactly see why this is needed.

Reply:
Because state space model test failed after the first commit. The model had 0 df_model obviously.:/

Reply:
In either case this cannot become the default option, and so must be explicitly passed by an end-user as an option to `test_serial_correlation`.

Reply:
So that means that even though the model instance contains the information on `df_model`, it nevertheless needs to be manually input by the end user?

Reply:
It can be automatic, but must be off by default.  Somehing like `df_adjust=False` as an additional keyword-only argument. If `df_adjust` is `False` then use `0`.  If `df_adjust` is `True`, then use an adjustment.  

I'm also not really sure if `df_model - 1` is always the correct choice.  Presumably, the `-1` is due to an assumption that the model contains an intercept-like term.  What if this isn't satisfied?

Reply:
Now I understand, thanks, I will modify the solution. Regarding the -1 problem, as far as I can see my provided example, it is functional even though it does not include an intercept (`trend='n'`), so I assume the -1 is due to residual standard error estimated coefficient.

Reply:
@bashtage I've updated the code according to your suggestion. If anything is not yet in order, please let me know.